pkg/pluginsupplier/plugin_supplier.go

```diff
@@ -27,62 +31,90 @@ func GetInstalledPlugins() ([]cli.PluginInfo, error) {
 
 // GetInstalledStandalonePlugins returns the installed standalone plugins.
 func GetInstalledStandalonePlugins() ([]cli.PluginInfo, error) {
-	standAloneCatalog, err := catalog.NewContextCatalog("")
+	standalonePlugins, _, err := getInstalledStandaloneAndServerPlugins()
 	if err != nil {
 		return nil, err
 	}
-	plugins := standAloneCatalog.List()
+	return standalonePlugins, nil
+}
 
-	// Any server plugin installed takes precedence over the same plugin
-	// installed as standalone.  We therefore remove those standalone
-	// plugins from the list.
-	plugins, err = removeInstalledServerPlugins(plugins)
+// GetInstalledServerPlugins returns the installed server plugins.
+func GetInstalledServerPlugins() ([]cli.PluginInfo, error) {
+	_, serverPlugins, err := getInstalledStandaloneAndServerPlugins()
 	if err != nil {
 		return nil, err
 	}
-	return plugins, nil
+	return serverPlugins, nil
 }
 
-// GetInstalledServerPlugins returns the installed server plugins.
-func GetInstalledServerPlugins() ([]cli.PluginInfo, error) {
-	serverNames, err := configlib.GetAllCurrentContextsList()
+func getInstalledStandaloneAndServerPlugins() (standalonePlugins, serverPlugins []cli.PluginInfo, err error) {
+	// Get all the standalone plugins found in the catalog
+	standAloneCatalog, err := catalog.NewContextCatalog("")
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
+	standalonePlugins = standAloneCatalog.List()
 
-	var serverPlugins []cli.PluginInfo
+	// Get all the server plugins found in the catalog
+	serverNames, err := configlib.GetAllCurrentContextsList()
+	if err != nil {
+		return nil, nil, err
+	}
 	for _, serverName := range serverNames {
 		if serverName != "" {
 			serverCatalog, err := catalog.NewContextCatalog(serverName)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 			serverPlugins = append(serverPlugins, serverCatalog.List()...)
 		}
 	}
 
-	return serverPlugins, nil
+	// If the same plugin is present both as standalone and
+	// as a server plugin we need to select which one to use
+	// based on the TANZU_CLI_STANDALONE_OVER_CONTEXT_PLUGINS variable
+	standalonePlugins, serverPlugins = mergeStandaloneAndServerPlugins(standalonePlugins, serverPlugins)
+	return standalonePlugins, serverPlugins, nil
 }
 
-// Remove any installed standalone plugin if it is also installed as a server plugin.
-func removeInstalledServerPlugins(standalone []cli.PluginInfo) ([]cli.PluginInfo, error) {
-	serverPlugins, err := GetInstalledServerPlugins()
-	if err != nil {
-		return nil, err
-	}
+// Remove an installed standalone plugin if it is also installed as a server plugin,
+// or vice versa if the TANZU_CLI_STANDALONE_OVER_CONTEXT_PLUGINS variable is enabled
+func mergeStandaloneAndServerPlugins(standalonePlugins, serverPlugins []cli.PluginInfo) (installedStandalone, installedServer []cli.PluginInfo) {
```

Comment:
`mergeStandaloneAndServerPlugins` seems confusing.
How about naming this function something like 
```
filterStandaloneAndServerPlugins(standalonePlugins, serverPlugins []cli.PluginInfo)
OR
filterStandaloneAndServerPluginsBasedOnPriority(standalonePlugins, serverPlugins []cli.PluginInfo)
OR
filterPluginsBasedOnPriority(standalonePlugins, serverPlugins []cli.PluginInfo)
OR
... (anything better)
```
?

Reply:
I did struggle looking for a proper name and in the end I gave up and took `mergeStandaloneAndServerPlugins`.
I like your `filter...` suggestions.  It made me think of a slight variation: `filterIdenticalStandaloneAndServerPlugins()`.
I'll use that and see how you feel about it.

Reply:
Yes. I like `filterIdenticalStandaloneAndServerPlugins` better. Let's use that.